app/calculator.js

```diff
@@ -0,0 +1,77 @@
+/**
+ *
+ * A simple class which performs the summarize and multiply actions
+ * @class Calculator
+ */
+
+class Calculator {
+  constructor() { }
+
+  /**
+   *
+   *
+   * @param {*} numbers is the first incoming value for the calculations
+   * @return {*} returns the sum result of number and increment
+   * @memberof Calculator
+   */
+  add(...numbers) {
+    if (numbers.length === 0) {
+      // eslint-disable-next-line max-len
+      throw new TypeError('Unable to calculate, number and increment are not defined');
+    } else if (numbers.length < 2 && typeof numbers[0] === 'number') {
+      throw new TypeError('Unable to calculate, only one number is defined');
+    } else if (numbers.length < 2 && typeof numbers[0] !== 'number') {
+      throw new TypeError(`Unable to calculate, [${numbers[0]}] is not a number`);
+    } else {
+      return numbers.reduce((sum, current) => {
+        if (typeof (current) !== 'number' || typeof (sum) !== 'number') {
+          // eslint-disable-next-line max-len
+          if (typeof (sum) !== 'number') {
+            // eslint-disable-next-line max-len
+            throw new TypeError(`Unable to calculate, [${sum}] is not a number`);
+          } else if (typeof (current) !== 'number') {
+            // eslint-disable-next-line max-len
+            throw new TypeError(`Unable to calculate, [${current}] is not a number`);
+          }
+        } else {
+          return sum + current;
+        }
+      }, 0);
+    }
+  }
```

Comment:
You've implemented too complicated logic here.
```
add(...numbers) {
    if (numbers.some((number) => typeof number !== 'number')) {
      const invalidArgument = numbers.find((arg) => typeof arg !== 'number');
      throw new Error(`Provided value ${invalidArgument} is not in the list of allowed data types`);
    } else {
      return numbers.reduce((previous, current) => previous + current);
    }
  }
  ```
  This variant is much better.

Reply:
This code does not work for cases when we ain't put nothing into a function or putting only one number:
In first case we're getting: 
'Reduce of empty array with no initial value'
In second case we're getting: 
'-calculator.add(...) is not a function'

That's a reason why the compacted logic was implemented